### ISSUE TITLE
Ability to set "privileged" flag (boolean) from environment var

### ISSUE THREAD

**Issue by brunocascio** [docker/compose#4028]
```
version: '2'
services:
  mongo:
    image: mongo
    privileged: $PRIVILEGED
```

But I got this error output:

```
ERROR: The Compose file './docker-compose.yml' is invalid because:
services.mongo.privileged contains an invalid type, it should be a boolean
```

Might be accept the value `"true"` will make sense..

Thanks

**Comment by mdantonio** [comment]
I were looking for a solution to the same problem (i would like to configure the privileged mode by using a variable in .env) and I found this issue.

Can I ask you why the pull request proposed to solve the problem (https://github.com/docker/compose/pull/4123) is closed? It has been rejected?

**Comment by shin-** [comment]
Implemented as part of #5291

**Comment by mdantonio** [comment]
It works, thank you! Now waiting for 1.18 to implement this in production (any plan about the release?)
